test/tests/functional/pbs_reliable_job_startup.py

```diff
@@ -5669,3 +5696,149 @@ def test_t21(self):
+        a = {'event': hook_event, 'enabled': 'true'}
+        # this hook increments job's select chunk request by 1
+        self.server.create_import_hook(hook_name, a, self.qjob_hook_body)
+
+        # test release_nodes() with execjob_prologue hook
+        hook_body = """
+import pbs
+e=pbs.event()
+pbs.logmsg(pbs.LOG_DEBUG, "Executing prologue")
+if e.job.in_ms_mom():
+    pj = e.job.release_nodes(keep_select=e.job.Resource_List["site"])
+    if pj != None:
+        pbs.logjobmsg(e.job.id, "launch: job.exec_vnode=%s" % (pj.exec_vnode,))
+    else:
+        e.job.delete()
+        e.reject("unsuccessful at PROLOGUE")
+"""
+        hook_event = "execjob_prologue"
+        hook_name = "prolo"
+        a = {'event': hook_event, 'enabled': 'true'}
+        self.server.create_import_hook(hook_name, a, hook_body)
+
+        jid = self.create_and_submit_job('job6')
+        # Job gets queued and reflects the incremented values from queuejob
+        # hook
+        self.server.expect(JOB, {'job_state': 'Q',
+                                 'tolerate_node_failures': 'job_start',
+                                 'Resource_List.mem': '6gb',
+                                 'Resource_List.ncpus': 9,
+                                 'Resource_List.nodect': 3,
+                                 'Resource_List.select': self.job6_iselect,
+                                 'Resource_List.site': self.job6_oselect,
+                                 'Resource_List.place': self.job6_place,
+                                 'schedselect': self.job6_ischedselect},
+                           id=jid, attrop=PTL_AND)
+
+        a = {'scheduling': 'true'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+
+        errmsg = \
+            "release_nodes(): not currently supported on Cray X* series nodes"
+        self.momA.log_match("Job;%s;%s" % (jid, errmsg), n=10)
+
+        self.momA.log_match("execjob_prologue request rejected by 'prolo'",
+                            n=10)
+
+        errmsg = "unsuccessful at PROLOGUE"
+        self.momA.log_match("Job;%s;%s" % (jid, errmsg), n=10)
+
+        self.server.expect(JOB, 'queue', op=UNSET, id=jid)
+
+        self.server.manager(MGR_CMD_DELETE, HOOK, id="prolo", expect=True)
+
+        # test release_nodes() with execjob_launch hook
+        hook_body = """
+import pbs
+e=pbs.event()
+
+if 'PBS_NODEFILE' not in e.env:
+    e.accept()
+
+pbs.logmsg(pbs.LOG_DEBUG, "Executing launch")
+if e.job.in_ms_mom():
+    pj = e.job.release_nodes(keep_select=e.job.Resource_List["site"])
+    if pj != None:
+        pbs.logjobmsg(e.job.id, "launch: job.exec_vnode=%s" % (pj.exec_vnode,))
+    else:
+        e.job.Hold_Types = pbs.hold_types("s")
+        e.job.rerun()
+        e.reject("unsuccessful at LAUNCH")
+"""
+        hook_event = "execjob_launch"
+        hook_name = "launch"
+        a = {'event': hook_event, 'enabled': 'true'}
+        self.server.create_import_hook(hook_name, a, hook_body)
```

Comment:
You successively add new hooks to test them in the same test case.  Won't the older hooks still run and reject the job first?